Ex4/Ex4.java

```diff
@@ -0,0 +1,81 @@
+public class Ex4 {
+
+	public static final boolean LAND = true;
+	public static final boolean WATER = false;
+	
+	public static boolean[][] map = {{WATER, LAND, WATER, LAND},
+					 {LAND, LAND, WATER, WATER},
+					 {WATER, WATER, LAND, WATER},
+					 {WATER, WATER, LAND, WATER}};
+	
+	public static void main(String[] args) {
+		System.out.print("Number of islands in the map: ");
+		System.out.println(countIslands(map));
+	}
+	
+	/**
+	 * iterate over each cell of the map.
+	 * when encountering land, mark all of its consecutive land neighbors
+	 * to form one island
+	 * @param map boolean map where true is land and false is water
+	 * @return the number of islands found
+	 */
+	public static int countIslands(boolean[][] map) {
+		int numberOfIslands = 0;
+		
+		for (int row = 0; row < map.length; row++) {
+			for (int col = 0; col < map[0].length; col++) {
+				if (map[row][col]) {
+					exploreNeighbors(row, col, map);
+					numberOfIslands++;
+				}
+			}
+		}
+		return numberOfIslands;
+	}
+
+	/**
+	 * Mark adjacent land tiles of specific cell coordinate in the map
+	 * as contiguous land to form an island and avoid counting it twice.
+	 * @param row
+	 * @param col
+	 * @param map
+	 */
```

Comment:
I took "Operating System" class just this semester, so I'm very familiar with the concepts you're describing, but not as much with implementing them. Will get to that.